src/lib/pubkey/mce/code_based_key_gen.cpp

```diff
@@ -10,25 +10,139 @@
  */
 
 #include <botan/internal/code_based_key_gen.h>
-#include <botan/code_based_util.h>
-#include <botan/gf2m_rootfind_dcmp.h>
-#include <botan/internal/binary_matrix.h>
+#include <botan/internal/code_based_util.h>
 #include <botan/loadstor.h>
-#include <botan/polyn_gf2m.h>
 
 namespace Botan {
 
 namespace {
 
+#define BITS_PER_U32 (8 * sizeof (u32bit))
```

Comment:
In the past there were a lot of strange computers (PDP endian, 28 bit longs, etc) which the C/C++ standards are often careful to allow, but is there any real (production, still supported and in use) computer where byte is not exactly 8 bits? From
https://stackoverflow.com/questions/2098149/what-platforms-have-something-other-than-8-bit-char it appears there are some DSPs which have a 16 bit char and no uint8_t type at all. But such a system could not be supported without a lot of work - all byte oriented math would be suspect and need examination and fixing.

But in this case this usage can go, because the code is probably allowed to assume that `u32bit` (aka `std::uint32_t`) is exactly 32 bits.